torch/csrc/jit/script/sugared_value.h

```diff
@@ -94,6 +94,32 @@ struct TORCH_API SugaredValue
     throw ErrorReport(loc) << "cannot call a " << kind();
   }
 
+  // If it is a iterable sugared value, use this interface to get the type information
+  // for all the iterables inside for loop, and prepare for loop information filling
+  virtual std::vector<TypePtr> getItersTypeInfo(
+    const SourceRange& loc,
+    Function& m) {
+   throw ErrorReport(loc) << kind() << " does not have the iterables type information";
```

Comment:
If these errors are user facing and need to describe what is happening in the same way python would if something could not be iterated.